site/app/libraries/DiffViewer.php

```diff
@@ -544,99 +544,104 @@ private function getDisplay($lines, $type = self::EXPECTED, $option = self::SPEC
          * there's a difference on that line or that the line doesn't exist.
          */
         $max_digits = strlen((string) count($lines));
-        for ($i = 0; $i < count($lines); $i++) {
-            $j = $i + 1;
-            if ($start === null && isset($this->diff[$type][$i])) {
-                $start = $i;
-                $html .= "\t<div class='highlight' id='{$this->id}{$type}_{$this->link[$type][$start]}'>\n";
-            }
-            if (isset($this->diff[$type][$i])) {
-                $html .= "\t<div class='bad'>";
-            }
-            else {
-                $html .= "\t<div>";
-            }
-            $html .= "<span class='line_number'>";
-            $digits_at_line = strlen((string) $j);
-            for ($counter = ($max_digits - $digits_at_line); $counter > 0; $counter--) {
-                $html .= "&nbsp;";
-            }
-            $html .= "{$j}</span>";
-            $html .= "<span class='line_code'>";
-            if (isset($this->diff[$type][$i])) {
-                // highlight the line
-                $current = 0;
-                // character highlighting
-                foreach ($this->diff[$type][$i] as $diff) {
-                    $html_orig = htmlentities(substr($lines[$i], $current, ($diff[0] - $current)));
-                    $test = str_replace("\0", "null", $html_orig);
-                    $html_orig_error = htmlentities(substr($lines[$i], $diff[0], ($diff[1] - $diff[0] + 1)));
-                    $test2 = str_replace("\0", "null", $html_orig_error);
-                    if ($option == self::SPECIAL_CHARS_ORIGINAL) {
-                        $html .= $html_orig;
-                        $html .= "<span class='highlight-char'>" . $html_orig_error . "</span>";
+        if (count($lines) > 5000) {
```

Comment:
We probably want to have this check above the insertion of blank lines above, so as to avoid the potential of a tons and tons of empty lines for specific diffs.

Also, please add a test case to https://github.com/Submitty/Submitty/blob/master/site/app/libraries/DiffViewer.php#L1 for these changes.

Reply:
Can you clarify your first suggestion or give an example? I am not following what you are saying.

Reply:
Before showing this too long message, the diff viewer does:

https://github.com/Submitty/Submitty/blob/fc17a81168719161270ca828ceb39d80d7ab5474/site/app/libraries/DiffViewer.php#L533-L541

which strikes me that it is possible to have a diff where it's some X number of empty lines followed with "This file is too long to display. Please download it instead.". We should probably not show these empty lines at all in the case that we are just going to show the too long message, especially given that the number of empty lines I think is unbounded?